metricbeat/module/kafka/partition/partition.go

```diff
@@ -37,57 +36,60 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch partition stats list from kafka
 func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 
-	if m.client == nil {
+	if m.broker == nil {
 		config := sarama.NewConfig()
 		config.Net.DialTimeout = m.Module().Config().Timeout
 		config.Net.ReadTimeout = m.Module().Config().Timeout
 		config.ClientID = "metricbeat"
 
-		client, err := sarama.NewClient([]string{m.Host()}, config)
+		broker := sarama.NewBroker(m.Host())
+		err := broker.Open(config)
 		if err != nil {
 			return nil, err
 		}
-		m.client = client
+		m.broker = broker
 	}
 
-	topics, err := m.client.Topics()
+	response, err := m.broker.GetMetadata(&sarama.MetadataRequest{})
```

Comment:
Topic filter "should" work here but I would keep this also for a version 2. It is already possible to filter topics out, but you would need `filters` which I think is ok for now.

For the question: TBH I'm not sure. Based on https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-TopicMetadataRequest I would expect that we would get an array for each broker. But we only get a list of topics back.